app/src/main/java/com/money/manager/ex/common/AllDataListFragment.java

```diff
@@ -232,28 +232,28 @@ public Loader<Cursor> onCreateLoader(int id, Bundle args) {
 //                    whereParams = whereParamsList.toArray(whereParams);
 //                }
 
-            // set sort
-            String sort = "";
-            if (args != null && args.containsKey(KEY_ARGUMENTS_SORT)) {
-                sort = args.getString(KEY_ARGUMENTS_SORT);
-            }
-
-            // create loader
             Dataset allData;
+            // create loader
             if (args != null && args.containsKey(ARG_SHOW_FLOATING_BUTTON)) {
                 // coming from report, use mobile data
                 allData = new QueryMobileData(getActivity());
             } else {
                 allData = new QueryAllData(getActivity());
             }
 
-            if ((new AppSettings(getContext())).getTransactionSort() == SORT_BY_DATE_DESC) {
-                sort = ((allData.getClass().equals(QueryAllData.class)) ? QueryAllData.Date : QueryMobileData.Date) + " DESC";
-            }
-            if ((new AppSettings(getContext())).getTransactionSort() == SORT_BY_DATE_ASC) {
-                sort = ((allData.getClass().equals(QueryAllData.class)) ? QueryAllData.Date : QueryMobileData.Date) + " ASC";
-            }
+            // set sort
+            String sort = "";
+            if (args != null && args.containsKey(KEY_ARGUMENTS_SORT)) {
+                sort = args.getString(KEY_ARGUMENTS_SORT);
+            } else {
+                if ((new AppSettings(getContext())).getTransactionSort() == SORT_BY_DATE_DESC) {
+                    sort = ((allData.getClass().equals(QueryAllData.class)) ? QueryAllData.Date : QueryMobileData.Date) + " DESC";
+                }
+                if ((new AppSettings(getContext())).getTransactionSort() == SORT_BY_DATE_ASC) {
+                    sort = ((allData.getClass().equals(QueryAllData.class)) ? QueryAllData.Date : QueryMobileData.Date) + " ASC";
+                }
 
+            }
 
```

Comment:
i didn't get what was changed here. i

Reply:
Introduced sort in report, but overwintering sort from intent. So a move all sort together.